File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/image_controller/ImageController.java
package coolpharaoh.tee.speicher.tea.timer.views.utils.image_controller;

import android.content.Intent;
import android.net.Uri;

import java.io.IOException;

public interface ImageController {
    Intent getSaveOrUpdateImageIntent(long teaId) throws IOException;

    Uri getImageUriByTeaId(long teaId);

    void removeImageByTeaId(long teaId);

    String getLastModified(Uri uri);
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/image_controller/ImageController.kt
package coolpharaoh.tee.speicher.tea.timer.views.utils.image_controller

import android.content.Intent
import android.net.Uri
import java.io.IOException

interface ImageController {

    @Throws(IOException::class)
    fun getSaveOrUpdateImageIntent(teaId: Long): Intent?

    fun getImageUriByTeaId(teaId: Long): Uri?

    fun removeImageByTeaId(teaId: Long)

    fun getLastModified(uri: Uri): String
}